### PR TITLE
Fix #148, with tests and other fixes

### LINKED ISSUE #148: Responsible groups are displayed in visible_for
When editing an event, responsible groups are displayed in the visible_for field although they were never added to visible_for. this is due to the fact that they are assigned the view_event permission in case they aren't in any groups that the event is visible for. We should exclude groups that are responsible from the initial set for the visible_for field

### PR DESCRIPTION
**Pull request by felixrindt**
Fix #148

### CODE REVIEW

File: ephios/event_management/forms.py [hunk 1]
```diff
@@ -46,6 +47,41 @@ class Meta:
         model = Event
         fields = ["title", "description", "location", "type", "mail_updates"]
 
+    def __init__(self, **kwargs):
+        user = kwargs.pop("user")
+        can_publish_for_groups = get_objects_for_user(user, "publish_event_for_group", klass=Group)
+
+        if (event := kwargs.get("instance", None)) is not None:
+            responsible_persons = get_users_with_perms(
```
Comment (jeriox) [inline ephios/event_management/forms.py:55]: should rename to responsible_users here as well

File: ephios/event_management/views.py [hunk 2]
```diff
@@ -63,51 +63,28 @@ class EventUpdateView(CustomPermissionRequiredMixin, UpdateView):
...
     def get_form(self, form_class=None):
-        visible_for_queryset = get_objects_for_user(
-            self.request.user, "publish_event_for_group", klass=Group
-        )
-        event_form = EventForm(
-            self.request.POST or None,
+        return EventForm(
+            data=self.request.POST or None,
+            user=self.request.user,
             initial={
```
Comment (jeriox) [inline ephios/event_management/views.py:80]: this is uneccessary as the form handles this now AFAIS?

File: ephios/event_management/forms.py [hunk 3]
```diff
@@ -66,12 +102,13 @@ def save(self, commit=True):
             Group.objects.filter(
                 Q(id__in=self.cleaned_data["visible_for"])
                 | Q(id__in=self.cleaned_data["responsible_groups"])
+                | Q(id__in=self.locked_visible_for_groups)
```
Comment (jeriox) [inline ephios/event_management/forms.py:105]: this is broken. steps to reproduce:
- have an event that is visible for volunteers and managers and planners are responsible
- visit the edit view with a user in the planner group
- you'll see volunteers in the select field for visible and "also for manager" in the help text
- remove volunteers from the visibility and save the page
- now the event is not visible for managers anymore

**Review by jeriox — CHANGES_REQUESTED** [review]
yay tests <3

File: ephios/event_management/forms.py (hunk 1 above)
Comment (jeriox) [reply ephios/event_management/forms.py:55]: the own user that is prefilled in initial still does not get saved/assigned change permissions. not sure if this is related to this renaming issue

**Review by jeriox — CHANGES_REQUESTED** [review]

File: ephios/event_management/views.py (hunk 2 above)
Comment (felixrindt) [reply ephios/event_management/views.py:80]: well no. this is the create view. the form only knows how to initialize stuff with a given instance.

File: ephios/event_management/forms.py (hunk 1 above)
Comment (felixrindt) [reply ephios/event_management/forms.py:55]: it probably was

File: ephios/event_management/forms.py (hunk 3 above)
Comment (felixrindt) [reply ephios/event_management/forms.py]: Very good catch. The locked groups were stored as an *unevaluated* queryset. Evaluation only happened after all permissions had been removed, so the queryset was empty. It is now initialized on form initialization.

**Review by jeriox — APPROVED** [review]

**Pull request merged**